src/others/minimax.js

```diff
@@ -0,0 +1,119 @@
+(function (exports) {
+  'use strict';
+  /* eslint max-params: 0 */
+
+  /**
+   * Minimax (sometimes MinMax, MM[1] or saddle point[2]) is a decision rule used in artificial intelligence,
+   * decision theory, game theory, statistics, and philosophy for minimizing the possible loss for a worst case (maximum loss) scenario.
+   * Optimized with alpha-beta pruning.
+   *  {@link https://en.wikipedia.org/wiki/Minimax}
+   *  {@link https://en.wikipedia.org/wiki/Alpha%E2%80%93beta_pruning}
+   *
+   * @public
+   * @module others/minimax
+   *
+   * @example
+   *
+   * var miniMax =
+   *  require('path-to-algorithms/src/others/minimax').minimax;
+   * var result = minimax(
+   *   [1, 2, 3],
+   *   true,
+   *   5,
+   *   -Infinity,
+   *   Infinity,
+   *   state => ({ move: 0, state: [2, 3, 4] }),
+   *   state => state[1] < 3,
+   *   state => state[1]
+   * );
+   *
+   * @param {*} state Current game state
+   * @param {Boolean} maximize Defines if the result should be maximized or minimized
+   * @param {Number} depth Defines the maximum depth search
+   * @param {Number} alpha Maximum score that the minimizing player is assured
+   * @param {Number} beta Minimum score that the maximizing player is assured
+   * @param {Function} getPossibleNextStatesFn Function which returns all possible next moves with states .
+   * @param {Function} isGameOverFn Function which returns if game is over.
+   * @param {Function} getScoreFn Function which returns score.
+   * @return {{score: Number, move: *}} which contains the minimum coins from the given
+   *  list, required for the change.
+   */
+  function minimax(
```

Comment:
There are some parameters which don't change over time:
- `getPossibleNextStatesFn`
- `isGameOverFn`
- `getScoreFn`

We can have a higher-order (builder) function, which accepts them and returns a new function, which does the rest:

```ts
const minimaxBuilder = (getPossibleNextStatesFn, isGameOverFn, getScoreFn) => {
  const minimax = (state, maximize, depth, alpha, beta) => {
    // ...
  };
  return minimax;
};
```

This way we'll change the public API a bit:

```ts
const minimax = minimaxBuilder(...);
const result = minimax(...);
```

Reply:
Great idea! Fixed.